### PR TITLE
Integrate travis-cookbooks changes plus tests

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

File: jdk_switcher.sh [hunk 1]
```diff
@@ -6,83 +6,94 @@ else
...
+    ORACLEJDK9_JAVA_HOME="/usr/lib/jvm/java-9-oracle-${ARCH_SUFFIX}"
 fi
 
-UJA="update-java-alternatives"
+for config_file in /etc/default/jdk-switcher "${HOME}/.jdk_switcherrc" "${JDK_SWITCHER_CONFIG}"; do
+    if [[ -f "${config_file}" ]]; then
+        # shellcheck source=/dev/null
+        source "${config_file}"
+    fi
+done
```
Comment (meatballhat) [inline jdk_switcher.sh]: Getting this documented would be a cool idea.  Is the `README` a decent place?

File: jdk_switcher.sh [hunk 2]
```diff
@@ -198,7 +219,7 @@ jdk_switcher() {
             print_java_home "$JDK"
             ;;
         *)
-            echo "Usage: $0 {use|home} [ JDK version ]" >&2
+            echo "Usage: jdk_switcher {use|home} [ JDK version ]" >&2
```
Comment (meatballhat) [inline jdk_switcher.sh]: The `$0` was resolving to something other than the `jdk_switcher` function name, which made sense to me, and I didn't research if there's a way in `bash` to get at the current context function name, so...

File: runtests [hunk 3]
```diff
@@ -3,11 +3,24 @@
...
+    "__runtests_${1:-lint}"
+}
+
+__runtests_help() {
+    echo 'Usage: runtests <lint|integration>'
+}
+
+__runtests_lint() {
     echo 'Running shellcheck'
-    git grep -l -E '^#!.+sh' | xargs shellcheck
```
Comment (meatballhat) [inline runtests]: The permissive pattern was picking up the `.bats` file, which is definitely not valid shell.

**Comment by meatballhat** [comment]
:heart: *bump*

**Comment by michaelklishin** [comment]
@meatballhat thank you! Sorry that it's been under my radar for almost a month.

**Pull request merged**